atom/browser/api/atom_api_desktop_capturer.cc

```diff
@@ -169,8 +169,14 @@ void DesktopCapturer::UpdateSourcesList(DesktopMediaList* list) {
       std::vector<std::string> device_names;
       // Crucially, this list of device names will be in the same order as
       // |media_list_sources|.
-      webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
-          &device_names);
+      bool success =
+          webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
+              &device_names);
+
+      if (!success)
```

Comment:
Shouldn't continue to iterate over screen captured sources for display id modifications once this has failed.

Reply:
fixed!